File: examples/azure/function/modular-easyauth/src/dotnet/HelloFunction.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace FuncEasyAuth;

public class HelloFunction(ILogger<HelloFunction> logger)
{
    [Function("hello")]
    public IActionResult Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = "hello")] HttpRequest req)
    {
        logger.LogInformation("hello function triggered");
        return new OkObjectResult(new { message = "hello-world" });
    }
}

File: examples/azure/function/modular-easyauth/src/java/src/main/java/com/example/HelloFunction.java
package com.example;

import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;

import java.util.Optional;

/**
 * Azure Functions HTTP trigger — hello-world with Anonymous auth level.
 * Easy Auth (Entra ID) is enforced at the App Service platform layer.
 */
public class HelloFunction {

    @FunctionName("hello")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET, HttpMethod.POST},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "hello"
            ) HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {

        context.getLogger().info("hello function triggered");

        return request.createResponseBuilder(HttpStatus.OK)
                .header("Content-Type", "application/json")
                .body("{\"message\":\"hello-world\"}")
                .build();
    }
}
